### ISSUE TITLE
Firefox -> OWR call initiation fails

### ISSUE THREAD

**Issue by ara4n** [EricssonResearch/openwebrtc#377]
Initiating calls from Firefox 37 to OWR fails with entirely blank video (using matrix dev webclient and dev iOS console), possibly due to some kind of VP8/H.264 negotiation mismatch?

**Comment by superdump** [comment]
Updated to tip of develop branch. Removed Pods and Podfile.lock. Ran pod install. Opened workspace, cleaned and built. Ran on 6+. Tested between Firefox 38 and iOS.

I did manage to get Firefox -> iOS to work one time but otherwise it would fail. It seems to be related to ICE. Sometimes Firefox would choose a mobile data IP on the phone side and claim that ICE had succeeded for that candidate but then no data would get through. I disabled mobile data temporarily to see if that helped and the next run was the one time that it worked.

However, on subsequent runs of initiating from Firefox with mobile data disabled, ICE would fail. I need to investigate the candidates being sent around to see what is going on. I can reproduce at least.

**Comment by superdump** [comment]
With the same TURN servers as are used in Matrix, this works fine between Firefox 38 and NativeDemo on iOS.

**Comment by stefanalund** [comment]
I think there is a difference in how we have the our TURN server configured. We currently enforce TCP, don't think matrix.org does that.

**Comment by ara4n** [comment]
yup, synapse's default coturn setup listens for turn on udp & tcp & tls. in practice I believe we see udp turn as very common (for NATs which don't allow inbound udp flows), plus it doesn't suffer from TCP windowing and congestion control latency problems, hence allowing it.

**Comment by superdump** [comment]
@stefanalund - perhaps I wasn't clear. I used our demo page with the matrix turn servers and our NativeDemo iOS app and the matrix turn servers and everything worked fine. There is something wrong in either the matrix web client or the iOS matrix console / owr SDK client.

As I was testing on a LAN, the turn servers shouldn't be needed at all but I wanted to replicate the setup.

**Comment by stefanalund** [comment]
@ara4n please correct me if I'm wrong, but I think you need a user specific password to access the Matrix TURN server. Could be that the server was not used in this test.

**Comment by superdump** [comment]
I pulled out the user and pass from the Matrix web client.

**Comment by superdump** [comment]
I think the next step is to compare the SDPs and candidates between demo.openwebrtc.io in FF -> NativeDemo and matrix.org beta web client in FF -> matrixConsole and play spot the difference. :)
